src/generateSpec.ts

```diff
@@ -153,15 +154,24 @@ export function getQueryParams(route: IRoute): oa.ParameterObject[] {
 
   const queriesMeta = _.find(route.params, { type: 'queries' })
   if (queriesMeta) {
-    const schema = getParamSchema(queriesMeta) as oa.ReferenceObject
-    queries.push({
-      in: 'query',
-      name: _.last(_.split(schema.$ref, '/')) || '',
-      required: isRequired(queriesMeta, route),
-      schema,
+    const paramSchema = getParamSchema(queriesMeta) as oa.ReferenceObject
+    const paramSchemaName = _.last(_.split(paramSchema.$ref, '/')) || ''
+    const schemas = validationMetadatasToSchemas({
```

Comment:
I'm not a huge fan of calling `validationMetadatasToSchemas` here and having `class-validator` as a dependency.

Can we instead find the query object among the schemas passed in via the [`additionalProperties` argument](https://github.com/epiphone/routing-controllers-openapi/blob/master/src/index.ts#L25)?

Reply:
Sure, I committed the changes. 

What do you think about this line?
 `const spec = getSpec(routes, additionalProperties.components?.schemas || {})`

Both components and schemas are optional, so I needed to provide a default value to keep TS happy. 

Gonna write some tests.

Reply:
@epiphone tests done

Reply:
That works well, I think.